### ISSUE TITLE
Incorrect error message for re-type-exporting built-ins

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#36167]
```ts
export type { number };
```

Expected: Cannot export 'number'. Only local declarations can be exported from a module.ts
Actual: 'number' only refers to a type, but is being used as a value here.
